### PR TITLE
feat: Update to flux 0.148.0

### PR DESCRIPTION
**Pull request by Marwes**
Rebased the branch I had prepared earlier on top of #374

Closes #374

### CODE REVIEW

File: src/completion.rs [hunk 1]
```diff
@@ -826,31 +824,42 @@ fn walk_package(
...
-                    push_var_result(&head.k, VarType::Duration)
-                }
-                MonoType::Regexp => {
-                    push_var_result(&head.k, VarType::Regexp)
-                }
-                MonoType::String => {
-                    push_var_result(&head.k, VarType::String)
-                }
-                _ => {}
+                MonoType::Arr(_) => push_var_result(
```
Comment (rockstar) [inline src/completion.rs:827]: Could we add a comment here to ensure that once Eq is implemented, we can un-nest this? Should we ever un-nest it? My preference is to have flat over nested, but I'm not sure if that preference holds up here.

File: src/server.rs [hunk 2]
```diff
@@ -611,7 +612,10 @@ impl LanguageServer for LspServer {
                 if let flux::semantic::nodes::Expression::Member(member) = callee.clone() {
                     let name = member.property.clone();
                     if let flux::semantic::nodes::Expression::Identifier(ident) = member.object.clone() {
-                        signatures.extend(find_stdlib_signatures(name, ident.name.to_string()));
+                        // XXX: rockstar (11 Jan 2022) - Symbol should be able to
+                        // be .into() String. When that happens, this `format!`
+                        // can go away.
+                        signatures.extend(find_stdlib_signatures(format!("{}", name), ident.name.to_string()));
```
Comment (rockstar) [inline src/server.rs:618]: Dang, I was hoping you already had a strategy in mind for addressing this. I'll fix this today.

**Review by rockstar — APPROVED** [review]
I shouldn't be able to fully review this patch, as I wrote some of it. Just had more comments for conversation and edification. Thanks for figuring out that issue I was bumping into. Very facepalm of me. :)

File: src/server.rs (hunk 2 above)
Comment (rockstar) [reply src/server.rs:618]: I have come around to using `.into()` where I would normally `.to_string()`, but that's mostly a personal preference. I think a followup PR with the change to the `find_stdlib_signatures` is the right thing.

**Comment by Marwes** [comment]
The `BuiltinType` change stem from changing `MonoType::Int/String/etc` to instead live under the new `BuiltinType` enum https://github.com/influxdata/flux/pull/4353 . There isn't any actual change in functionality.

File: src/server.rs (hunk 2 above)
Comment (Marwes) [reply src/server.rs:618]: Yah, I use `into()` quite a bit as well for `String`, though it may fail inference sometimes which can be annoying

**Review by scbrickley — APPROVED** [review]

**Pull request merged**